WordPress build.sh: npm install lacks --legacy-peer-deps and hides errors via --silent (breaks React 18/19 block repos)

## Summary

The WordPress build extension's `build.sh` runs `npm install` / `npm ci` **without `--legacy-peer-deps`** and **with `--silent`**, so it cannot build any plugin/theme whose dependency tree has a React 18 vs 19 peer split (i.e. anything pulling both `@wordpress/blocks@^15.x` and `@wordpress/block-editor@^14.x` / `@extrachill/components`). The install hard-fails with `ERESOLVE`, but `--silent` swallows the error, so `homeboy build` just prints `Installing npm dependencies...` and exits with an opaque `success: false` and no visible cause.

## Repro

`homeboy build extrachill-community --path <worktree>` on a repo whose `package.json` includes:

```json
"@wordpress/blocks": "^15.9.0",        // peer: react@^19.2.4 (resolves to 15.20)
"@wordpress/block-editor": "^14.0.0",  // peer: react@^18
"@extrachill/components": "^0.5.2"     // peer: react@^18 || ^19
```

homeboy output stops at:

```
[BUILD] Detected @wordpress/scripts build tool
[BUILD] Installing npm dependencies...
{ "success": false, "data": { "command": "build.run", "success": false } }
```

The underlying npm error (hidden by `--silent`) is:

```
npm error code ERESOLVE
npm error While resolving: extrachill-community@1.2.3
npm error Found: react@18.3.1 (peer from @wordpress/block-editor@14, @extrachill/components)
npm error Could not resolve dependency:
npm error   peer react@"^19.2.4" from @wordpress/blocks@15.20.0
npm error Conflicting peer dependency: react@19.2.6
npm error Fix the upstream dependency conflict, or retry
npm error this command with --force or --legacy-peer-deps
```

A bare `npm install` in the same dir reproduces it directly (RC=1, no `node_modules` created).

## Why it's not obvious

`extrachill-analytics` builds fine through homeboy because it has **no** peer conflict — it depends only on `@wordpress/api-fetch` + `@wordpress/element` + `wp-native-client`, no `@wordpress/blocks`/`block-editor`/`components`. So the gap only bites repos that actually compose block-editor Gutenberg blocks (the majority of EC block/theme repos).

## Affected install location

`~/.config/homeboy/extensions/wordpress/scripts/build/build.sh` (byte-identical to the repo copy at `wordpress/scripts/build/build.sh`, so this is current, not a stale install):

```sh
# install_frontend_dependencies()
if [ -f "package-lock.json" ]; then
    npm ci --silent --no-audit --no-fund 2>&1
else
    npm install --silent --no-audit --no-fund 2>&1
fi
print_success "${scope_label}npm dependencies ready"
```

Two distinct bugs here:

1. **No `--legacy-peer-deps`** → install fails for any React 18/19 peer-split tree. This is the standard, expected flag for `@wordpress/scripts` projects on modern npm (npm 7+ enforces peers). The `wp-scripts` ecosystem broadly assumes it.
2. **`--silent` + unconditional `print_success`** → the ERESOLVE (or any install error) is hidden, and `print_success "npm dependencies ready"` prints even when the install failed. The build then dies later (or reports an opaque failure) with no actionable signal. The install step should surface stderr on failure and not claim success when npm returned non-zero.

## Suggested fix

- Add `--legacy-peer-deps` to both the `npm ci` and `npm install` invocations (or make it configurable via an env var, defaulting on for `@wordpress/scripts` projects).
- Drop `--silent` (or capture output and echo it on non-zero exit), and gate `print_success` on the npm exit code so a failed install fails loudly instead of printing "dependencies ready".

## Notes

- Disk/inodes are healthy on the affected host (56G free, 30% inodes); this is purely a dependency-resolution flag gap, not a resource issue.
- Confirmed reproducible: `npm install --legacy-peer-deps` succeeds where bare `npm install` fails, in the exact same worktree.

--- a/wordpress/scripts/build/build-helper-smoke.sh
+++ b/wordpress/scripts/build/build-helper-smoke.sh
@@ -76,6 +76,6 @@ JSON
         bash "${EXTENSION_DIR}/scripts/build/build.sh" >/dev/null
 )
 
-assert_contains "$npm_log" "install --silent --no-audit --no-fund --legacy-peer-deps"
+assert_contains "$npm_log" "install --no-audit --no-fund --legacy-peer-deps"
 
 echo "WordPress build helper smoke passed."

--- a/wordpress/scripts/build/build.sh
+++ b/wordpress/scripts/build/build.sh
@@ -358,7 +358,7 @@ install_frontend_dependencies() {
     elif [ -f "package-lock.json" ] && [ "package-lock.json" -nt "node_modules/.package-lock.json" ]; then
         print_warning "${scope_label}package-lock.json is newer than node_modules. Reinstalling dependencies..."
         need_install=1
-    elif [ -f "package-lock.json" ] && ! npm ls --depth=0 --silent >/dev/null 2>&1; then
+    elif [ -f "package-lock.json" ] && ! npm ls --depth=0 $(npm_install_flags) >/dev/null 2>&1; then
         print_warning "${scope_label}node_modules contains missing or invalid packages. Reinstalling dependencies..."
         need_install=1
     elif [ -n "$expected_bin" ] && [ ! -x "node_modules/.bin/$expected_bin" ]; then
@@ -368,9 +368,9 @@ install_frontend_dependencies() {
 
     if [ "$need_install" -eq 1 ]; then
         if [ -f "package-lock.json" ]; then
-            run_npm_install ci --silent 2>&1
+            run_npm_install ci 2>&1
         else
-            run_npm_install install --silent 2>&1
+            run_npm_install install 2>&1
         fi
         print_success "${scope_label}npm dependencies ready"
     fi

